pkg/controllers/binding/common.go

```diff
@@ -319,3 +335,72 @@ func shouldSuspendDispatching(suspension *policyv1alpha1.Suspension, targetClust
 	}
 	return suspendDispatching
 }
+
+func updateBindingDispatchingConditionIfNeeded(ctx context.Context, client client.Client, eventRecorder record.EventRecorder, binding client.Object, scope apiextensionsv1.ResourceScope) error {
+	newBindingSchedulingCondition := metav1.Condition{
+		Type:               workv1alpha2.Suspended,
+		LastTransitionTime: metav1.Now(),
+	}
+
+	bindingSuspendScheduling := false
+	var conditions *[]metav1.Condition
+	switch scope {
+	case apiextensionsv1.NamespaceScoped:
+		bindingObj := binding.(*workv1alpha2.ResourceBinding)
+		bindingSuspendScheduling = util.IsBindingSuspendScheduling(bindingObj)
+		conditions = &bindingObj.Status.Conditions
+	case apiextensionsv1.ClusterScoped:
+		bindingObj := binding.(*workv1alpha2.ClusterResourceBinding)
+		bindingSuspendScheduling = util.IsClusterBindingSuspendScheduling(bindingObj)
+		conditions = &bindingObj.Status.Conditions
+	default:
+		klog.ErrorS(nil, "Unsupported resource binding scope")
+		return nil
+	}
+
+	if bindingSuspendScheduling {
+		newBindingSchedulingCondition.Status = metav1.ConditionTrue
+		newBindingSchedulingCondition.Message = SuspendedSchedulingConditionMessage
+		newBindingSchedulingCondition.Reason = SuspendedSchedulingConditionReason
+	} else {
+		// There is no need to set an extra Suspended condition when scheduling not suspended and Suspended condition not exists.
+		if meta.FindStatusCondition(*conditions, workv1alpha2.Suspended) == nil {
+			return nil
+		}
```

Comment:
Looks like a bug here, losing the chance to update the condition from `Suspended-true` to `Suspended-false`.

Reply:
This check only retrun when `Suspended` condition is not existing, and if `Suspended` condition is true, it will be updated to false in the code below: )

Reply:
OK, got it.